File: Basic Maths/LCM of two numbers.cpp
class Solution {
public:
    int LCM(int n1,int n2) {
        int lcm;
        int i = 1;
        int maxnum = max(n1, n2);
        while(i)
        {
            int multiple = i * maxnum;
            if(multiple % n1 == 0 && multiple % n2 ==0)
            {
                lcm = multiple;
                break;
            }
            i = i + 1;

        }
        
       
        return lcm;

    }
};

File: Basic Maths/LCM of two numbers.py
"""
📝 Question:
Given two positive integers a and b, find their LCM (Least Common Multiple),
i.e., the smallest number that is a multiple of both a and b.

Example:
  Input:  a = 12, b = 8
  Output: 24
"""

# LCM (Least Common Multiple) of Two Numbers

def gcd(a, b):
    while b:
        a, b = b, a % b
    return a

def lcm(a, b):
    return (a * b) // gcd(a, b)


if __name__ == "__main__":
    print(lcm(12, 8))   # 24
    print(lcm(5, 7))    # 35
